fixed error for regexp search in 'annotate--font-lock-matcher'

cage2 opened the pull request:
Sometimes some modes/package puts overlay  on the last character of a
  buffer  (notably  SLIME when  the  parenthesis  of  a form  are  not
  balanced). This will make  're-search-forward' in the aforementioned
  function  fails  and font  lock  becomes  a  mess (e.g.  text  color
  disappears).

Commit:
- fixed error for regexp search in 'annotate--font-lock-matcher'

  Sometimes some modes/package puts overlay  on the last character of a
  buffer  (notably  SLIME when  the  parenthesis  of  a form  are  not
  balanced). This will make  're-search-forward' in the aforementioned
  function  fails  and font  lock  becomes  a  mess (e.g.  text  color
  disappears).

Commit:
- fixed version number.

bastibe commented:
Wow, great find. The font-lock code is a bit hairy. I know I spent a long time initially to figure out how it worked.

Happy new year, by the way!

cage2 commented:
On Mon, Jan 06, 2020 at 12:09:06AM -0800, Bastian Bechtold wrote:

Hi Bastian!

> Wow, great find. The font-lock code is a bit hairy.

I agree! ;-)

[...]

>
> Happy new year, by the way!

Thank you Bastian! I wish a happy new year to you too! :)

Bye!
C.

cage2 commented:
On Mon, Jan 06, 2020 at 01:49:59AM -0800, cage2 wrote:

Hello Bastian!

Have you got some spare time to give your opinion about this patch?  I
am using it  on my system and  seems that there are  no regressions or
such. I wonder if could be merged.

Bye!
C.

bastibe commented:
Sorry about that. Feel free to contact me much earlier, if I'm not responding, I had forgotten about this.

If you have the time, please remove the double spaces in the comment. Otherwise, the patch looks good!

Commit:
- removed multiple, consecutive, spaces in comments.

Commit:
- increased version number;
- updated NEWS;
- created Changelog.

cage2 commented:
On Wed, Jan 22, 2020 at 01:56:43AM -0800, Bastian Bechtold wrote:

Hello Bastian!

> Sorry about that.

No need to apologize at all! :)

> Feel free to  contact me much earlier, if I'm  not responding, I had
  forgotten about this.

This happens to everyone sometimes, no problem at all! :)

>
> If you have the time, please remove the double spaces in the comment. Otherwise, the patch looks good!

I have removed  the consecutive spaces (there was actually  a lot! :-(
). Also i have increased the  version number, updated NEWS and added a
changelog like we discussed early. I hope everything is fine, if yes i
am going to merge this patch tomorrow (next 24 hours).

Bye!!!
C.

bastibe commented:
Great! Thanks a lot for another wonderful contribution!

cage2 commented:
On Wed, Jan 22, 2020 at 12:17:51PM -0800, Bastian Bechtold wrote:
> Great! Thanks a lot for another wonderful contribution!

You're welcome!  Honestly  if not for SLIME mode that  uses overlays a
lot i never discovered this bug, in other words: i was just lucky! :D

Bye!
C.
